File: day2/first.py
with open("input.txt") as file:
    lines = file.read().splitlines()
    result = 0
    loss = ['A Z', 'B X', 'C Y']
    win = ['A Y', 'B Z', 'C X']
    for line in lines:
        if line in loss:
            pass
        elif line in win:
            result += 6
        else:
            result += 3
        shape = line[-1]
        match shape:
            case 'X':
                result += 1
            case 'Y':
                result += 2
            case 'Z':
                result += 3
    print(result)

File: day2/first.rb
file = File.open('input.txt')
lines = file.read.split("\n")
result = 0
lost = ['A Z', 'B X', 'C Y']
win = ['A Y', 'B Z', 'C X']
shape_point = { 'X': 1, 'Y': 2, 'Z': 3 }
lines.each do |line|
  if lost.include?(line)
    nil
  elsif win.include?(line)
    result += 6
  else
    result += 3
  end
  # @type [String]
  shape = line[-1]
  result += shape_point[shape.to_sym]
end
puts result

file.close
